src/auth/__tests__/LoginWrapper.test.tsx

```diff
@@ -83,4 +83,46 @@ describe("redirects when user becomes authenticated", () => {
       shallow: true
     });
   });
+
+  test("uses homepage when nextUrl start with http", () => {
+    render(<LoginWrapper />, {
+      user: {
+        isAuthenticated: true
+      },
+      router: {
+        query: {
+          nextUrl: "http://test.com/test123"
+        }
+      }
+    });
+
+    expect(mockPush).toHaveBeenCalledTimes(1);
+    // redirects to home page instead of nextUrl
+    expect(mockPush).toHaveBeenCalledWith("/testlib", undefined, {
+      shallow: true
+    });
+  });
+
+  test("uses nextUrl when url contains http", () => {
```

Comment:
Minor: Might be slightly confusing. Maybe "uses nextUrl when url contains, but does not start with, http".